libs/shared/src/consts/handlebar-helpers/getTemplateVariables.ts

```diff
@@ -46,6 +46,10 @@ export function getTemplateVariables(bod: any[]): IMustacheVariable[] {
         };
       });
 
+      if (['with'].includes(body.path.original)) {
+        return [...nestedVariablesInBlock];
+      }
```

Comment:
[with block helper](https://handlebarsjs.com/guide/builtin-helpers.html#with) is an object, not an array as we have treated it so far. 
{{#with person}} {{firstName}} {{lastName}} {{/with}}
should render variables as person.firstName person.lastName